### PR TITLE
Feature/wt 6153

### PR DESCRIPTION
**Pull request by alvinllobrera**

### CODE REVIEW

File: src/main/java/io/wisetime/connector/ServerRunner.java [hunk 1]
```diff
@@ -125,7 +125,8 @@ public Server getServer() {
     private WiseTimeConnector wiseTimeConnector;
     private ApiClient apiClient;
     private String apiKey;
-    private TemplateFormatterConfig.Builder templateConfigBuilder = TemplateFormatterConfig.builder();
+    private TemplateFormatterConfig.Builder publicTemplateConfigBuilder = TemplateFormatterConfig.builder();
+    private TemplateFormatterConfig.Builder internalTemplateConfigBuilder = TemplateFormatterConfig.builder();
```
Comment (alvinllobrera) [inline src/main/java/io/wisetime/connector/ServerRunner.java]: Patricia uses another template to create charge record

File: src/main/java/io/wisetime/connector/utils/TagDurationCalculator.java [hunk 2]
```diff
@@ -30,18 +31,33 @@ public static double tagDuration(final TimeGroup timeGroup) {
...
+   * Calculate the tag duration to apply to each tag.
+   *
+   * @param timeGroup the TimeGroup for which to calculate each tag duration
+   * @param useTimeGroupDuration if True, the method will use the time group's total duration to calculate tag duration,
+   *                             otherwise, it will use the sum of the duration of each time row in the time group.
+   * @param disregardExperienceRating if True, it will apply the user experience rating to calculate tag duration, otherwise,
+   *                                  it will disregard user's experience rating.
+   * @return duration, in seconds, to apply to each tag
+   */
+  public static double tagDuration(final TimeGroup timeGroup,
```
Comment (alvinllobrera) [inline src/main/java/io/wisetime/connector/utils/TagDurationCalculator.java]: Patricia uses the total duration of group time rows to record actual hours worked in Time Registration

File: src/main/java/io/wisetime/connector/ServerRunner.java (hunk 1 above)
Comment (alvinllobrera) [reply src/main/java/io/wisetime/connector/ServerRunner.java]: Patricia have different content on comment/narrative for Time Registration and Budget Line record.
Budget Line record contains additional data (e.g user experience factor)

File: src/main/java/io/wisetime/connector/ServerRunner.java [hunk 3]
```diff
@@ -208,24 +210,49 @@ public ServerBuilder withApiClient(ApiClient apiClient) {
...
      */
-    public ServerBuilder withTemplatePath(String templatePath) {
-      templateConfigBuilder.withTemplatePath(templatePath);
+    public ServerBuilder withPublicTemplatePath(String templatePath) {
+      publicTemplateConfigBuilder.withTemplatePath(templatePath);
+      return this;
+    }
+
+    /**
+     * Comparing to the {@link ServerBuilder#ServerRunner#withPublicTemplatePath(String)} this
```
Comment (alvinllobrera) [inline src/main/java/io/wisetime/connector/ServerRunner.java]: Added notes for the distinction of a public template and an internal template

File: src/main/java/io/wisetime/connector/ServerRunner.java [hunk 4]
```diff
@@ -208,24 +210,49 @@ public ServerBuilder withApiClient(ApiClient apiClient) {
...
+      publicTemplateConfigBuilder.withMaxLength(maxLength);
       return this;
     }
 
     /**
      * @see TemplateFormatterConfig#DEFAULT_MAX_LENGTH
      */
-    public ServerBuilder withTemplateMaxLength(int maxLength) {
-      templateConfigBuilder.withMaxLength(maxLength);
+    public ServerBuilder withInternalTemplateMaxLength(int maxLength) {
```
Comment (alvinllobrera) [inline src/main/java/io/wisetime/connector/ServerRunner.java]: Allows specific connector to specify a template path for an internal template

File: src/main/java/io/wisetime/connector/integrate/ConnectorModule.java [hunk 5]
```diff
@@ -34,7 +47,11 @@ public ApiClient getApiClient() {
     return apiClient;
   }
 
-  public TemplateFormatter getTemplateFormatter() {
-    return templateFormatter;
+  public TemplateFormatter getPublicTemplateFormatter() {
+    return publicTemplateFormatter;
+  }
+
+  public TemplateFormatter getInternalTemplateFormatter() {
```
Comment (alvinllobrera) [inline src/main/java/io/wisetime/connector/integrate/ConnectorModule.java]: Connector should call this to generate an internal comment/narrtive

**Comment by vyshane** [comment]
Per conversation in Slack, we have added new `DurationCalculator`, and have decided to manually instantiate extra `TemplateFormatter`s for now. I'll close this PR.

**Pull request closed without merging**